src/engineio/socket.rs

```diff
@@ -1,22 +1,86 @@
 #![allow(unused)]
 use std::{ops::Deref, thread};
 
-use super::packet::{Packet, PacketId, HandshakePacket};
-use crate::engineio::transport::TransportClient;
+use crate::engineio::packet::{decode_payload, encode_payload, Packet, PacketId};
 use crate::error::{Error, Result};
-use bytes::Bytes;
+use adler32::adler32;
+use bytes::{BufMut, Bytes, BytesMut};
 use native_tls::TlsConnector;
-use reqwest::header::HeaderMap;
-use std::sync::{
-    atomic::{AtomicBool, Ordering},
-    Arc, RwLock,
+use reqwest::{
+    blocking::{Client, ClientBuilder},
+    header::HeaderMap,
+    Url,
 };
+use serde::{Deserialize, Serialize};
+use std::{borrow::Cow, time::SystemTime};
+use std::{fmt::Debug, sync::atomic::Ordering};
+use std::{
+    sync::{atomic::AtomicBool, Arc, Mutex, RwLock},
+    time::{Duration, Instant},
+};
+use websocket::{
+    client::sync::Client as WsClient,
+    header::Headers,
+    sync::stream::{TcpStream, TlsStream},
+    ClientBuilder as WsClientBuilder,
+};
+use websocket::{
+    dataframe::Opcode, receiver::Reader, sync::Writer, ws::dataframe::DataFrame, Message,
+};
```

Comment:
Can't we union those imports? Or those the linter then starts to complain?

Reply:
done.